# <div style="text-align: center;">Assignment-07(Pandas)</div>

## Description of the dataset:
* **AQI_data.csv is a dataset showing AQI values and pollution category based on the AQI values for different cities of Bangladesh.**
* **AQI values and pollution category of each city are presended side by side.**

<h1>Importing Libraries 

In [ ]:
import pandas as pd
import numpy as np


# Reading Data

In [ ]:
df = pd.read_csv('./AQI_data.csv')
df

Date  Dhaka Pollution Category  Chattogram Pollution Category2  \
0     5/9/2024    158          UNHEALTHY         101             CAUTION   
1     5/9/2024    158          UNHEALTHY         101             CAUTION   
2    5/11/2024    100           MODERATE         103             CAUTION   
3    5/11/2024    100           MODERATE         103             CAUTION   
4     3/8/2024    166          UNHEALTHY         104             CAUTION   
..         ...    ...                ...         ...                 ...   
263   4/3/2024    137            CAUTION          93                 NaN   
264   4/8/2024    165          UNHEALTHY         207                 NaN   
265   4/8/2024    165          UNHEALTHY         207                 NaN   
266  5/18/2024    117            CAUTION          82                 NaN   
267  5/18/2024    117            CAUTION          83                 NaN   

    Gazipur Pollution Category3 Narayanganj Pollution Category4 Sylhet  ...  \
0       198           UNHEALTHY         140             CAUTION    112  ...   
1       198           UNHEALTHY         140             CAUTION    112  ...   
2       116             CAUTION         105             CAUTION     66  ...   
3       116             CAUTION         105             CAUTION     66  ...   
4       186           UNHEALTHY         246      Very Unhealthy    DNA  ...   
..      ...                 ...         ...                 ...    ...  ...   
263     DNA                 NaN         151           UNHEALTHY     61  ...   
264     197           UNHEALTHY         DNA                 NaN    175  ...   
265     197           UNHEALTHY         DNA                 NaN    175  ...   
266     151           UNHEALTHY          88            MODERATE    107  ...   
267     151           UNHEALTHY          88            MODERATE    107  ...   

    Savar Pollution Category9 Mymensingh  Pollution Category10 Cumilla  \
0     143             CAUTION        178             UNHEALTHY     173   
1     143             CAUTION        178             UNHEALTHY     173   
2     DNA                 NaN        156             UNHEALTHY     160   
3     DNA                 NaN        156             UNHEALTHY     160   
4     201      Very Unhealthy        247        Very Unhealthy     254   
..    ...                 ...        ...                   ...     ...   
263   159           UNHEALTHY        205        Very Unhealthy     155   
264   174           UNHEALTHY        182             UNHEALTHY     194   
265   174           UNHEALTHY        182             UNHEALTHY     194   
266   110             CAUTION        150               CAUTION     154   
267   110             CAUTION        150               CAUTION     154   

    Pollution Category11 Norshingdi Pollution Category12 Rangpur  \
0              UNHEALTHY        DNA                  NaN     204   
1              UNHEALTHY        DNA                  NaN     204   
2              UNHEALTHY        DNA                  NaN     169   
3              UNHEALTHY        DNA                  NaN     169   
4         Very Unhealthy        175            UNHEALTHY     183   
..                   ...        ...                  ...     ...   
263            UNHEALTHY        144              CAUTION     242   
264            UNHEALTHY        177            UNHEALTHY     220   
265            UNHEALTHY        177            UNHEALTHY     220   
266            UNHEALTHY        178            UNHEALTHY     160   
267            UNHEALTHY        178            UNHEALTHY     160   

    Pollution Category13  
0         Very Unhealthy  
1         Very Unhealthy  
2              UNHEALTHY  
3              UNHEALTHY  
4              UNHEALTHY  
..                   ...  
263       Very Unhealthy  
264       Very Unhealthy  
265       Very Unhealthy  
266            UNHEALTHY  
267            UNHEALTHY  

[268 rows x 27 columns]

# Analyze DataFrames

# Describe the dataset

# Find Nulls values in each column

# How many pollution category is used to describe air quality based on the AQI value?(Check in multiple districts) 

# How many pollution category is used to describe air quality based on the AQI value?

# How many times did each pollution category occur in Dhaka?

# Rename the column name of Pollution Category to Pollution_Dhaka

# Rename the column name of Pollution Category 2 to Pollution_Chattogram

# Rename the column name of Pollution Category 2 to Pollution_Gazipur

# How many times air quality of Dhaka was healthy?

# How many times air quality of Dhaka was unhealthy?

# How many times air quality of Dhaka was very unhealthy?

# How many times air quality of Dhaka was good?

# How many times air quality of Dhaka was moderate?

# How many times air quality of Dhaka was unhealthy or very unhealthy?

# How many times AQI value of Dhaka was above 100 ?

# How many times AQI value of Chattogram was above 100 ?

# How many times AQI value of Dhaka was between 100 and 150?

# What is the minimum and maximum AQI value of Dhaka?

# What is the minimum and maximum AQI value of Chattogram?

# What is the mean AQI value of Chattogram?

# What is the mean AQI value of Dhaka?